src/test/java/fakes/UserControllerTest.java

```diff
@@ -28,54 +35,127 @@ void withValidInexistingUsername_returnsOK__FAKE() {
       // 1. Test schneller machen
       // 2. UserController.create so beinflussen,
       //    dass einmal der "if"- und einmal der "else"-Fall durchlaufen wird
+
+      UserController ctrl = new UserController(new FakeUserValidator(false));
+      User user = new User("peter");
+
+      Message result = ctrl.create(user);
+
+      Assertions.assertEquals(result.status, Message.Status.OK);
+    }
+
+    @Test
+    void withValidExistingUsername_returnsNOK__FAKE() {
+      // TODO
+      // 1. Test schneller machen
+      // 2. UserController.create so beinflussen,
+      //    dass einmal der "if"- und einmal der "else"-Fall durchlaufen wird
+
+      UserController ctrl = new UserController(new FakeUserValidator(true));
+      User user = new User("kalua");
+
+      Message result = ctrl.create(user);
+
+      Assertions.assertEquals(result.status, Message.Status.NOT_OK);
     }
 
+
     @Test
     void withValidInexistingUsername_returnsOK__MOCKITO() {
       // TODO
+
+      UserValidator uv = mock(UserValidator.class);
+      doReturn(true).when(uv).doesUsernameExist(anyString());
+
+      Assertions.assertTrue(uv.doesUsernameExist("Foo"));
+
     }
 
     @Test
     void withValidInexitingUsername_addUserToDB__FAKE() {
       // TODO
       // Der Test soll prüfen, ob der Benutzer tatsächlich der DB hinzugefügt wurde.
       // Dazu soll ein Database Mock Objekt verwendet werden.
+
+      MockDatabase database = new MockDatabase();
+      UserController ctrl = new UserController(new FakeUserValidator(false), database);
+      User user = new User("peter");
+
+      Message result = ctrl.create(user);
+
+      Assertions.assertEquals(result.status, Message.Status.OK);
+      boolean doesUserExist = false;
+      for(User u : database.getUsers()){
+        if (u.getUsername().equals("peter")){
+          doesUserExist = true;
+        }
+      }
+      Assertions.assertTrue(doesUserExist);
     }
 
     @Test
     void withValidInexitingUsername_addUserToDB__MOCKITO() {
       // TODO
-    }
+      // Add User ist eine void method! -> kein return wert - ich fake einen Aufruf mit assertion true aber macht für mich keinen sinn...
+      Database database = new MockDatabase();
```

Comment:
Versuche in diesem Test nur Mockito verwenden, keine selbst erstellten Mock Klassen.

Ebenfalls sollte das System Under Test (SUT) hier der UserController sein. Du hast die Database und den UserValidator gemocked. Das wäre ja ok, aber dir fehlt der Aufruf des UserControllers.

Also es müsste so ungefähr aussehen: `...; new UserController(...).create(user);  ... verify(..., times(1))...`

Die Assertion sollte sicherstellen, dass database.addUser(user) aufgerufen wurde, genau einmal.